Fix JSON deserialization errors in C# 'REST Client' tutorial. Closes …

mozhganEtaati opened the pull request:
…#49387

Fix the deserialization issue in the Repository model.

## Summary

Corrected property names in the Repository model to match the JSON fields returned by the GitHub API.

GitHubHomeUrl → maps to html_url

LastPushUtc → maps to pushed_at

Fixes #49387 


<!-- PREVIEW-TABLE-START -->

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/tutorials/console-webapiclient.md](https://github.com/dotnet/docs/blob/12a03a93a6c769803a57bed17a512cac7434f085/docs/csharp/tutorials/console-webapiclient.md) | [Tutorial: Make HTTP requests in a .NET console app using C\#](https://review.learn.microsoft.com/en-us/dotnet/csharp/tutorials/console-webapiclient?branch=pr-en-us-49442) |


<!-- PREVIEW-TABLE-END -->

mozhganEtaati committed:
Fix JSON deserialization errors in C# 'REST Client' tutorial. Closes #49387

Fix the deserialization issue in the Repository model.

mozhganEtaati committed:
Refactor Repository class to use record definition

Updated the Repository class definition to use a record type with additional properties for JSON deserialization.

mozhganEtaati committed:
Add JSON property names to Repository record

mozhganEtaati committed:
Remove LastPush property from model

Removed LastPush property and its implementation.

mozhganEtaati committed:
Update console-webapiclient.md

StuartMosquera commented on docs/csharp/tutorials/console-webapiclient.md:
I think it would be helpful to give some context of the reason to import the `System.Text.Json.Serialization` namespace and the use of `JsonPropertyName` attribute.

mozhganEtaati committed:
Update documentation text for Repository section

mozhganEtaati committed:
Change step numbers

mozhganEtaati committed:
Update console-webapiclient.md for JSON deserialization

Clarified JSON property name handling and added note on case insensitivity of GetFromJsonAsync.

BillWagner reviewed (approved):
Thanks @mozhganEtaati 
And @StuartMosquera 

This looks great, and I'll :shipit: now.

BillWagner commented on docs/csharp/tutorials/console-webapiclient.md:
````diff
@@ -209,26 +209,28 @@ The `ProcessRepositoriesAsync` method can do the async work and return a collect
-    ```
+In the following steps, we extend the code to process more properties from the JSON payload returned by the GitHub API. You probably won't need to process every property, but adding a few demonstrates additional C# features.
+
+1. Replace the contents of the `Repository` class with the following `record` definition. Make sure to import the `System.Text.Json.Serialization` namespace and apply the `[JsonPropertyName]` attribute to map JSON fields to C# properties explicitly.
````
Add a blank line for lint:

```suggestion
1. Replace the contents of the `Repository` class with the following `record` definition. Make sure to import the `System.Text.Json.Serialization` namespace and apply the `[JsonPropertyName]` attribute to map JSON fields to C# properties explicitly.

```

BillWagner reviewed (commented):
I spoke too soon. There's a couple lint issues to address. I've added suggestions that I think will fix them.

mozhganEtaati committed:
Update docs/csharp/tutorials/console-webapiclient.md

Co-authored-by: Bill Wagner <wiwagn@microsoft.com>

mozhganEtaati committed:
Update docs/csharp/tutorials/console-webapiclient.md

Co-authored-by: Bill Wagner <wiwagn@microsoft.com>

mozhganEtaati commented:
@BillWagner 
Hello,
I noticed the PR hasn’t been merged yet.
Is there any issue or blocker that I should look into?
